pytest/lib/bridge.py

```diff
@@ -150,51 +277,45 @@ def start(self):
                 self.config_dir,
                 'logs/eth2near-relay/err.log'),
             'a')
-        cli_dir = os.path.join(self.bridge_dir, 'cli')
-        args = ('node index.js start eth2near-relay --daemon false').split()
-        self.pid.value = subprocess.Popen(
-            args, stdout=self.stdout, stderr=self.stderr, cwd=cli_dir).pid
-        # TODO ping and wait until service really starts
-        time.sleep(5)
-        assert self.pid.value != 0
-
-
-class JSAdapter:
-
-    def __init__(self, config):
-        self.config = config
-        self.bridge_dir = self.config['bridge_dir']
-        self.cli_dir = os.path.join(self.bridge_dir, 'cli')
-
-    def call(self, args):
-        if not isinstance(args, list):
-            args = [args]
-        args = ['node', 'index.js', 'TESTING'] + args
-        # TODO check for errors
-        return subprocess.check_output(
-            args, cwd=self.cli_dir).decode('ascii').strip()
+        self.pid.value = self.adapter.js_call_service('eth2near-relay', self.stdout, self.stderr).pid
+        if self.pid.value == 0:
+            logger.error('Cannot start Eth2NearBlockRelay')
+            assert False
 
 
-def assert_deployed(output):
-    assert output.decode('ascii').strip().split(
-        '\n')[-1].strip().split(' ')[0] == 'Deployed'
+def assert_success(message):
+    try:
+        message = message.decode('ascii')
+    except:
+        # message is already str
+        pass
+    if message == '':
+        # empty message is considered valid
+        return
+    message = message.strip().split('\n')[-1].strip().split(' ')
+    if not (message[0] in ('Deployed', 'Done') or message[-1] in ('deployed', 'initialized')):
+        logger.error(message)
+        assert False
 
 
 class RainbowBridge:
 
-    def __init__(self, config):
+    def __init__(self, config, node):
         self.config = config
+        self.node = node
         self.eth2near_block_relay = None
         self.near2eth_block_relay = None
         self.adapter = JSAdapter(self.config)
         self.bridge_dir = self.config['bridge_dir']
         self.config_dir = self.config['config_dir']
         self.cli_dir = os.path.join(self.bridge_dir, 'cli')
         if not os.path.exists(self.bridge_dir):
-            self._git_clone_install()
+            self.git_clone_install()
         if not os.path.exists(os.path.expanduser("~/go")):
-            print('Go must be installed')
+            logger.error('Go must be installed')
             assert False
+        # TODO resolve
+        # This hack is for NayDuck
```

Comment:
Why is this hack required?

Reply:
Previously it was `cp`. Pytest folder and Bridge folder appear on different filesystems at Nayduck and it's not allowed to `cp` files between them. Debugging this is extremely time-consumable because each run at Nayduck takes a lot of time and gives small knowledge for reverse-engineering.

I've found that running `wget` ignores the issue with filesystems. I suggest to patch this hack separately of this PR under @utka consulting, if it will be necessary for some reason.

Reply:
Interesting. This is not a big problem, and actually we probably don't need to resolve anything here.